src/sparsezoo/objects/directories.py

```diff
@@ -302,48 +302,31 @@ def __init__(self, *args, download_alias: Optional[str] = None, **kwargs):
         self.download_alias = download_alias
 
     @contextmanager
-    def _override_name(self):
+    def _download_with_alias(self):
         """
         A context manager to temporarily override the name of the directory
         to the download alias.
         """
         old_name = self.name
+        old_url = self.url
+        old_is_archive = self.is_archive
         if self.download_alias:
             self.name = self.download_alias
+            self.is_archive = True
+            self.url = self.get_file(self.download_alias).url
         yield
         self.name = old_name
+        self.url = old_url
+        self.is_archive = old_is_archive
 
     def download(self, *args, **kwargs):
         """
         Override the download method to temporarily override the name
         of the directory to the download alias.
         """
-        with self._override_name():
-            return super().download(*args, **kwargs)
-
-    @property
-    def path(self):
-        """
-        Override the path property to download and extract
-        the download alias file but point to the path of the
-        actual expected file
-
-        :raises FileNotFoundError: if the expected file does not
-            exist, (it wasn't downloaded or extracted)
-        :return: path to the expected file
-        """
-        super().path
-        actual_file_path = (
-            Path(self._path).with_name(self.name)
-            if self._path.endswith(self.download_alias)
-            else Path(self._path)
-        )
-        if not actual_file_path.exists():
-            raise FileNotFoundError(
-                f"The directory {actual_file_path} should have been "
-                "downloaded but does not exist"
-            )
-        return str(actual_file_path)
+        with self._download_with_alias():
+            super().download(*args, **kwargs)
+            self.unzip()
```

Comment:
why are we moving this functionality here? this shouldn't be a requirement/side effect of an "aliased" directory